src/components/Settings/DatasourceSettings/index.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import ListItem from 'react-md/lib/Lists/ListItem';
+import List from 'react-md/lib/Lists/List';
+import plugins from '../../../data-sources/plugins';
+
+interface IDatasourceSettingsProps {
+  settings: IDataSourceContainer;
+}
+
+interface IDatasourceSettingsState { 
+  selectedIndex: number;
+}
+
+export default class DatasourceSettings extends React.Component<IDatasourceSettingsProps, IDatasourceSettingsState> {
+
+  static uniqueId = 0;
+
+  state = {
+    selectedIndex: 0
+  };
+    
+  constructor(props: IDatasourceSettingsProps) {
+    super(props);
+
+    this.onMenuClick = this.onMenuClick.bind(this);
+    this.getSettings = this.getSettings.bind(this);
+    this.renderSettingsEditor = this.renderSettingsEditor.bind(this);
+    this.renderMenuItems = this.renderMenuItems.bind(this);
+    this.getSettingsEditor = this.getSettingsEditor.bind(this);
+  }
+  
+  getSettingsEditor(element: DataSource): JSX.Element {
+
+    if (!element) { return null; }
+
+    let ReactElementClass = plugins[element.type];
+    if (ReactElementClass && ReactElementClass.editor) {
+      let SettingsEditor = ReactElementClass.editor;
+      return <SettingsEditor key={DatasourceSettings.uniqueId++} settings={element} />;
+    }
+    return null;
+  }
+
+  onMenuClick(index: number) {
+    this.setState({ selectedIndex: index });
+  }
+
+  getSettings() {
+    return this.props.settings.dataSources.length > this.state.selectedIndex ?
+        this.props.settings.dataSources[this.state.selectedIndex] :
+        null;
+  }
+  
+  renderSettingsEditor() {
```

Comment:
Should return a : JSX.Element (or something similar)
HAE where should return a react element